File: pset1/mario1/less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //initialize i
    int height;
    //do-while loop always runs at least once
    do
    {
        //prompt user for an integer
        height = get_int("Height: ");
    }

    //do that for when the integer is (< 0 or > 23) --> "invalid"
    while (height < 0 || height > 23);

     // for loop iterates through ROWS which is the height
     // contains 2 nested routes
    for (int row = 0; row < height; row++)
    { //for loop to create correct amount of spaces for given# of bricks
        for (int space = 0; space < height - row -  1; space++)
        {
            printf(" ");
        } // for loop to create correct # of hashes
        for (int hashes = 0; hashes < row + 2; hashes++)
        {
            printf("#");
        } //to put in the proper line breaks after each loop
        printf("\n");
    }
}

File: pset1/mario1/less/mario.py
# program that takes an integer as an input
# and outputs a pyramid in mario form
# the pyramid has 'height' number of rows and the size of each row decreases by one at each step

from cs50 import get_int

while True:
    # always prompt user for input first
    height = get_int("Input integer for height: ")
    # if user input IS in range
    if height >= 0 and height <= 23:
        print("The height is: ", height)
        break
    elif height < 0 and height > 23:
        # reprompt the user
        print("Int must be between 0 and 23")
        height = get_int("Input integer for height: ")
        break

# for each loop iterates through height int
# ex/ 4 --> 1, 2, 3, 4
for i in range(height):
    # print spaces * (height - i)
    # print hashtags * (i + 2)
    print(" " * (height - i) + "#" * (i + 2))






